CMakeLists.txt

```
@@ -159,9 +168,15 @@ if(${KALDI_USE_PATCH_NUMBER})
     set(KALDI_VERSION "${KALDI_VERSION}.${KALDI_PATCH_NUMBER}")
 endif()
 
-get_third_party(openfst)
-set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
-include(third_party/openfst_lib_target)
+# get_third_party(openfst)
+# set(OPENFST_ROOT_DIR ${CMAKE_BINARY_DIR}/openfst)
+# include(third_party/openfst_lib_target)
+find_library(OpenFST
+            NAMES fst 
+            PATHS ${CMAKE_SOURCE_DIR}/tools/openfst/lib 
+            REQUIRED)
+include_directories(${CMAKE_SOURCE_DIR}/tools/openfst/include)
+link_directories(${CMAKE_SOURCE_DIR}/tools/openfst/lib)

```

```suggestion
link_directories(${OpenFST_LIBRARY_DIR})
```
This is a _cached_ path to the library. Once found, the `find_library()` command won't perform the search again. Also, use `OpenFST_INCLUDE_DIRS` for the `include_directories()`. It's tempting to omit `link_directories()` here altogether and use `link_libraries(${OpenFST_LIBRARY})`, which contains a full path to the library, but it will be incorrect if someone toggles the `BUILD_SHARED_LIBS` value: the cached value retains the .a or .so suffix which was selected according to the value of that variable.